### PR TITLE
fix: Get cost for `TransactionReceiptQuery` and `AccountBalanceQuery`

### LINKED ISSUE #990: Investigate `TransactionReceiptQuery` execution flow
### Description

`TransactionReceiptQuery` does not call `getCost` in it's execution flow. Investigate if this is expected behaviour and if not - normalise it compared to other queries.

### Steps to reproduce

-

### Additional context

_No response_

### Hedera network

other

### Version

2.40.0

### Operating system

None

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:
`TransactionReceiptQuery` and `AccountBalanceQuery ` do not call getCost in it's execution flow.
Make it's flow similar to `TransactionRecordQuery`.

**Related issue(s)**:
The branch name should be "receipt" instead of "record".

Fixes #990

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: transaction_receipt_query_unit_test.go [hunk 1]
```diff
@@ -150,6 +150,7 @@ func TestUnitTransactionPlatformNotActiveGracefulHandling(t *testing.T) {
 			},
 		},
 	}}
+
```
Comment (0xivanov) [inline transaction_receipt_query_unit_test.go:153]: Remove

File: transaction_record_query_unit_test.go [hunk 2]
```diff
@@ -251,6 +251,19 @@ func TestUnitTransactionRecordReceiptNotFound(t *testing.T) {
 				},
 			},
 		},
+		&services.Response{
```
Comment (0xivanov) [inline transaction_record_query_unit_test.go:254]: Remove

File: query.go [hunk 3]
```diff
@@ -215,7 +215,7 @@ func (q *Query) execute(client *Client, e QueryInterface) (*services.Response, e
 	}
 
 	var cost Hbar
-	if q.queryPayment.tinybar == 0 {
+	if q.queryPayment.tinybar == 0 && q.isPaymentRequired {
```
Comment (0xivanov) [inline query.go:218]: There are 2 queries that don't require payment: `AccountBalance` and `Receipt`

File: transaction_receipt_query.go [hunk 4]
```diff
@@ -102,40 +101,17 @@ func (q *TransactionReceiptQuery) GetCost(client *Client) (Hbar, error) {
...
-
-	q.pb = q.buildQuery()
-
-	if q.isPaymentRequired && len(q.paymentTransactions) > 0 {
-		q.paymentTransactionIDs._Advance()
-	}
-	q.pbHeader.ResponseType = services.ResponseType_ANSWER_ONLY
-
-	resp, err := _Execute(client, q)
+	resp, err := q.Query.execute(client, q)
```
Comment (0xivanov) [inline transaction_receipt_query.go:104]: Do the same for `AccountBalance`

**Review by gsstoykov — APPROVED** [review]
LGTM.

**Pull request merged**